Docs: Update documentation for 0.13.1 release

amogh-jahagirdar opened the pull request:
In this change we are copying over the new markdowns and doc changes from the Iceberg repo. Validated by running hugo serve for the landing page/docs.

jackye1995 commented on landing-page/layouts/partials/header.html:
```diff
@@ -39,6 +39,9 @@
                     <a class="page-scroll" href="{{ .Site.BaseURL }}community">Community</a>
                 </li>
+                <li>
+                    <a class="page-scroll" href="{{ .Site.BaseURL }}contribute">Contribute</a>
```
I think it's worth adding a contribute tab, but how many tabs can we add in header? I hope we don't keep adding new tabs and cause issue in UI. @samredai any guideline here we should follow?

samredai replied:
I know some people have suggested drop-downs, which I think could work but I personally like the simplicity of just having the items there. I don't think we need to have top level links to every page. What if `/contribute` is only reachable by a link included in `community.md` and also a link in the iceberg repo's readme?

jackye1995 replied:
have a link in community page also works for me.

jackye1995 reviewed (approved):
overall looks good to me! @samredai let us know what you think about adding the new tab for contributing.

samredai commented on docs/content/docs/integrations/aws.md:
```diff
@@ -480,7 +480,7 @@ AWS_PACKAGES=(
 
 ICEBERG_PACKAGES=(
-  "iceberg-spark-runtime-3.2_2.12"
+  "iceberg-spark3-runtime"
```
Looks like these changes were made here but not synced back in `apache/iceberg`. @jackye1995 should we do one more manual resolution and try to get a consensus on a contributing pattern?

jackye1995 replied:
yes agree, let's do manual resolutions for these for 0.13.1, and let me start a thread for doc change contribution pattern in the dev list

samredai commented on docs/content/docs/spark/spark-procedures.md:
```diff
@@ -275,9 +275,9 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
-See the [`RewriteDataFiles` Javadoc](../../../javadoc/{{% icebergVersion %}}/org/apache/iceberg/actions/RewriteDataFiles.html#field.summary),
-<br/>  [`BinPackStrategy` Javadoc](../../../javadoc/{{% icebergVersion %}}/org/apache/iceberg/actions/BinPackStrategy.html#field.summary)
-and <br/> [`SortStrategy` Javadoc](../../../javadoc/{{% icebergVersion %}}/org/apache/iceberg/actions/SortStrategy.html#field.summary)
+See the [`RewriteDataFiles` Javadoc](./javadoc/{{ versions.iceberg }}/org/apache/iceberg/actions/RewriteDataFiles.html#field.summary),
```
Here as well, the `../../../javadoc/*` are the correct links.

samredai commented on landing-page/content/common/format/spec.md:
```diff
@@ -34,15 +34,15 @@ The format version number is incremented when new features are added that will b
 Version 1 of the Iceberg spec defines how to manage large analytic tables using immutable file formats: Parquet, Avro, and ORC.
 
-All version 1 data and metadata files are valid after upgrading a table to version 2. [Appendix E](spec/#version-2) documents how to default version 2 fields when reading version 1 metadata.
+All version 1 data and metadata files are valid after upgrading a table to version 2. [Appendix E](#version-2) documents how to default version 2 fields when reading version 1 metadata.
```
This should be kept as well, `spec/#version-2` is the correct link.

samredai commented on landing-page/content/common/format/spec.md:
```diff
@@ -839,7 +839,7 @@ Note that the string map case is for maps where the key type is a string. Using
 Values should be stored in Parquet using the types and logical type annotations in the table below. Column IDs are required.
 
-Lists must use the [3-level representation](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists).
+Lists must use the [3-level representation](https://github.com/apache/parquet-format/blob/master/LogicalTypes#lists).
```
The original link here should be kept.

samredai commented:
@amogh-jahagirdar sorry there are a few changes that are being introduced but are actually reversions. This is because right as the new site was released there were a number of changes that were being introduced in both places and weren't synced. I think this should all stable out over time but for not I think you can just remove the changes to those lines in this PR (I think I've tagged all of them expect the `spec/*` ones which are sprinkled on that page.

@jackye1995 I think once this is merged, I can do one more full copy from here to `apache/iceberg` and we should be ready to tighten up the contribution rules.

amogh-jahagirdar commented:
> @amogh-jahagirdar sorry there are a few changes that are being introduced but are actually reversions. This is because right as the new site was released there were a number of changes that were being introduced in both places and weren't synced. I think this should all stable out over time but for now I think you can just remove the changes to those lines in this PR (I think I've tagged all of them expect the `spec/*` ones which are sprinkled on that page.

No problem! I will remove all the relevant changes. After this is merged, we can copy back to iceberg.

amogh-jahagirdar commented:
Updated, removed the relevant changes with the bad links (I think we can sync back to iceberg after this gets merged as @samredai  mentioned). I also made a contributing section in "Join" which links to the new contributing page.

jackye1995 reviewed (approved):
checked all the inconsistencies pointed out are fixed

jackye1995 commented:
merging this to continue with the release process, we can add patch in case anything is wrong, thanks for the review @samredai and thanks for the work @amogh-jahagirdar!
